pkg/apis/serving/v1beta1/component.go

```diff
@@ -218,6 +223,10 @@ type PodMetricSource struct {
 	Target MetricTarget `json:"target"`
 }
 
+type ExtMetricAuth struct {
+	Name string `json:"name"`
+}
```

Comment:
another config we need is the `authModes`  https://keda.sh/docs/2.17/scalers/prometheus/#authentication-parameters

Reply:
Isn't it already there?
https://github.com/kserve/kserve/pull/4384/files#diff-803846cd158b42171183a3183847297969dc1fe6d895881c5fd941e367dcd7b3R206

Reply:
yes, this comment is a month old